Source/gg2/Objects/Weapons/Revolver.events/User Event 1.xml

```diff
@@ -19,11 +19,11 @@
     {
         var seed;
         seed = irandom(65535);
-        sendEventFireWeapon(ownerPlayer, seed);
+        sendEventFireWeapon(ownerPlayer, seed);                            
         doEventFireWeapon(ownerPlayer, seed);
     }
 }
-else if(readyToStab &amp;&amp; owner.cloak)
+else if(readyToStab &amp;&amp; owner.cloak &amp;&amp; !mouse_check_button(mb_right))
```

Comment:
Don't hardcode the mouse button. GG2 has configurable controls.

Also, this may not work properly in multiplayer.